nvim/session.lua

```diff
@@ -175,7 +175,12 @@ function Session:_run(request_cb, notification_cb, timeout)
       self._prepare:stop()
     end)
   end
-  self._msgpack_rpc_stream:read_start(request_cb, notification_cb, uv.stop)
+  self._msgpack_rpc_stream:read_start(request_cb, notification_cb, function()
+    uv.run()  -- run the loop to get exitcode from child process.
```

Comment:
Not really sure about `uv.run()` here - `uv.run('nowait')` also works in my test, but might not be enough always?